test/main.ml

```diff
@@ -29,23 +31,30 @@ let test_file f =
 
 let test_directory d =
   let count_error = ref 0 in
-  Format.printf "testing directory: `%a`@." Fpath.pp d;
+  Format.printf "testing directory     : `%a`@." Fpath.pp d;
   match Bos.OS.Dir.contents ~rel:false d with
   | Ok l ->
     List.iter
       (fun file ->
         incr count_total;
-        match test_file file with
-        | Ok () -> ()
+        match test_file ~optimize:false file with
+        | Ok () -> begin
+          incr count_total;
+          match test_file ~optimize:true file with
```

Comment:
You're running this test only if the previous one was successful. It would be better to run it in all cases.